opensquirrel/exceptions.py

```diff
@@ -0,0 +1,20 @@
+"""This module contains all custom exception used by ``OpenSquirrel``."""
+
+from typing import Any
+
+
+class UnsupportedGateError(Exception):
+    """Should be raised when a gate is not supported."""
+
+    def __init__(self, gate: Any, *args: Any) -> None:
+        """Init of the ``UnsupportedGateError``.
+
+        Args:
+            gate: Gate that is not supported.
+        """
+        self.unsupported_gate = gate
+        super().__init__(f"The following gate is not supported: {gate}.", *args)
+
```

Comment:
```suggestion
        super().__init__(f"the following gate is not supported: {gate}.", *args)

```

Error messages should start in lowercase, because they will usually be suffixed (with `Error: `, or `Cannot export circuit: `, etc.)